graspologic/pipeline/embed/omnibus_embedding.py

```diff
@@ -0,0 +1,285 @@
+          to handle sparse matrices that may have large slowly decaying spectrum.
+      svd_seed : Optional[int] (default=None)
+          Used to seed the PRNG used in the ``randomized`` svd solver algorithm.
+      weight_attribute : str (default="weight")
+          The edge dictionary key that contains the weight of the edge.
+
+      Returns
+      -------
+      List[Tuple[Embeddings, Embeddings]]
+
+      Raises
+      ------
+      beartype.roar.BeartypeCallHintPepParamException if parameters do not match type hints
+      ValueError if values are not within appropriate ranges or allowed values
+
+      See Also
+      --------
+      graspologic.pipeline.embed.Embeddings
+      graspologic.embed.OmnibusEmbed
+      graspologic.embed.AdjacencySpectralEmbed
+      graspologic.embed.select_svd
+
+      References
+      ----------
+      .. [1] Levin, K., Athreya, A., Tang, M., Lyzinski, V., & Priebe, C. E. (2017,
+         November). A central limit theorem for an omnibus embedding of multiple random
+         dot product graphs. In Data Mining Workshops (ICDMW), 2017 IEEE International
+         Conference on (pp. 964-967). IEEE.
+
+      .. [2] Sussman, D.L., Tang, M., Fishkind, D.E., Priebe, C.E.  "A
+         Consistent Adjacency Spectral Embedding for Stochastic Blockmodel Graphs,"
+         Journal of the American Statistical Association, Vol. 107(499), 2012
+
+      .. [3] Levin, K., Roosta-Khorasani, F., Mahoney, M. W., & Priebe, C. E. (2018).
+          Out-of-sample extension of graph adjacency spectral embedding. PMLR: Proceedings
+          of Machine Learning Research, 80, 2975-2984.
+
+      .. [4] Zhu, M. and Ghodsi, A. (2006). Automatic dimensionality selection from the
+          scree plot via the use of profile likelihood. Computational Statistics & Data
+          Analysis, 51(2), pp.918-930.
+    """
+    check_argument(len(graphs) > 1, "more than one graph is required")
+
+    check_argument(dimensions >= 1, "dimensions must be positive")
+
+    check_argument(elbow_cut is None or elbow_cut >= 1, "elbow_cut must be positive")
+
+    check_argument(
+        svd_solver_algorithm in __SVD_SOLVER_TYPES,
+        f"svd_solver_algorithm must be one of the values in {','.join(__SVD_SOLVER_TYPES)}",
+    )
+
+    check_argument(svd_solver_iterations >= 1, "svd_solver_iterations must be positive")
+
+    check_argument(
+        svd_seed is None or 0 <= svd_seed <= 2 ** 32 - 1,
+        "svd_seed must be a nonnegative, 32-bit integer",
+    )
+
+    weight_attribute = _graphs_precondition_checks(graphs, weight_attribute)
+
+    graph_embeddings = []
+
+    # create a graph that contains all nodes and edges across the entire corpus
+    union_graph = graphs[0].copy()
+    for graph in graphs[1:]:
+        union_graph.add_edges_from(graph.edges())
+
+    union_graph_lcc = largest_connected_component(union_graph)
+    union_graph_lcc_nodes = union_graph_lcc.nodes()
+    union_node_ids = np.array(list(union_graph_lcc_nodes))

```

Comment:
just wanted to mention that we have a util function for this here https://github.com/microsoft/graspologic/blob/07c78921491da596e9f438c740805c7a920aaa0d/graspologic/utils/utils.py#L584

Reply:
@nyecarr maybe you can use this in #824 too?